### PR TITLE
Simulation Gazebo: Use Gazebo Airpressure sensor

### PR DESCRIPTION
**Pull request by ahcorde**
Added Airpressure sensor that allows to setup the barometer in Gazebo

FYI @Jaeyoung-Lim @bperseghetti @dagar

### CODE REVIEW

**Comment by dagar** [comment]
I've merged https://github.com/PX4/PX4-Autopilot/pull/21136, can you rebase?

**Comment by ahcorde** [comment]
> I've merged #21136, can you rebase?

@dagar Done

File: ROMFS/px4fmu_common/init.d-posix/px4-rc.simulator [hunk 1]
```diff
@@ -86,10 +86,6 @@ elif [ "$PX4_SIMULATOR" = "gz" ] || [ "$(param show -q SIM_GZ_EN)" -eq "1" ]; th
 
 		# start gz bridge with pose arg.
 		if gz_bridge start -p "${model_pose}" -m "${PX4_GZ_MODEL}" -w "${PX4_GZ_WORLD}" -i "${px4_instance}"; then
-			if param compare -s SENS_EN_BAROSIM 1
-			then
-				sensor_baro_sim start
-			fi
```
Comment (Jaeyoung-Lim) [inline ROMFS/px4fmu_common/init.d-posix/px4-rc.simulator:92]: I think this should stay, but the sensor sim should be disabled in the airframe config. Set  the parameter to zero
https://github.com/PX4/PX4-Autopilot/blob/b66e15c4b967a144b193d6b71f4f0b315e7631f0/ROMFS/px4fmu_common/init.d-posix/airframes/4001_gz_x500#L17

File: src/modules/simulation/gz_bridge/GZBridge.cpp [hunk 2]
```diff
@@ -329,28 +337,52 @@ void GZBridge::clockCallback(const gz::msgs::Clock &clock)
...
 	const uint64_t time_us = (air_pressure.header().stamp().sec() * 1000000)
 				 + (air_pressure.header().stamp().nsec() / 1000);
 
-	double air_pressure_value = air_pressure.pressure();
+	// publish
+	sensor_baro_s sensor_baro{};
+	sensor_baro.timestamp_sample = time_us;
+	sensor_baro.device_id = 6620172; // 6620172: DRV_BARO_DEVTYPE_BAROSIM, BUS: 1, ADDR: 4, TYPE: SIMULATION
+	sensor_baro.pressure = air_pressure.pressure();
+	sensor_baro.temperature = this->_temperature;
```
Comment (Jaeyoung-Lim) [inline src/modules/simulation/gz_bridge/GZBridge.cpp:355]: Does this mean that the temperature field is missing from the message itself?

File: src/modules/simulation/gz_bridge/GZBridge.hpp [hunk 3]
```diff
@@ -94,7 +95,8 @@ class GZBridge : public ModuleBase<GZBridge>, public ModuleParams, public px4::S
 
 	void clockCallback(const gz::msgs::Clock &clock);
 
-	//void airpressureCallback(const gz::msgs::FluidPressure &air_pressure);
+	// void airspeedCallback(const gz::msgs::AirSpeedSensor &air_pressure);
+	void airpressureCallback(const gz::msgs::FluidPressure &air_pressure);
```
Comment (Jaeyoung-Lim) [inline src/modules/simulation/gz_bridge/GZBridge.hpp:99]: Would it be possible to change the naming to something less confusing with the airspeed (e.g. `barometerCallback` ?)

File: src/modules/simulation/gz_bridge/GZBridge.cpp (hunk 2 above)
Comment (ahcorde) [reply src/modules/simulation/gz_bridge/GZBridge.cpp:355]: the temperature field is not available in the fluid pressure message https://github.com/gazebosim/gz-msgs/blob/gz-msgs9/proto/gz/msgs/fluid_pressure.proto

File: src/modules/simulation/gz_bridge/GZBridge.hpp (hunk 3 above)
Comment (ahcorde) [reply src/modules/simulation/gz_bridge/GZBridge.hpp]: https://github.com/PX4/PX4-Autopilot/pull/21176/commits/e0de6492a0bb545b661adcd0234d21856aa25c26

File: ROMFS/px4fmu_common/init.d-posix/px4-rc.simulator (hunk 1 above)
Comment (ahcorde) [reply ROMFS/px4fmu_common/init.d-posix/px4-rc.simulator]: https://github.com/PX4/PX4-Autopilot/pull/21176/commits/e0de6492a0bb545b661adcd0234d21856aa25c26

File: src/modules/simulation/gz_bridge/GZBridge.cpp (hunk 2 above)
Comment (Jaeyoung-Lim) [reply src/modules/simulation/gz_bridge/GZBridge.cpp:357]: Not related to this PR, but would it make sense to add it?

**Review by Jaeyoung-Lim — APPROVED** [review]

**Pull request merged**